app/models/maths_and_physics/eligibility.rb

```diff
@@ -106,6 +107,15 @@ def initial_teacher_training_specialised_in_maths_or_physics?
       itt_subject_maths? || itt_subject_physics? || itt_specialism_physics?
     end
 
+    # Returns a String representing the answer given for the QTS question when
+    # the claim was made.
+    def qts_award_year_answer
```

Comment:
I agree that this is definitely a presentation method. I don‘t feel very comfortable with it being in the model (and off the top of my head I can‘t think of a precedent for this in the codebase).

If the aim of this is to have presentation logic that can be shared between multiple presenters, you could consider adding a `PresenterMethods` module which is shared by all presenters (or put it in a helper and include it in the presenters). We already have an `Admin::PresenterMethods` module so there’s precedent for it.

Reply:
Yeah, I prevaricated a lot about introducing this method but felt on balance the system is safer and more easily worked with having it here and not elsewhere.

I don't believe the aim is to _"have presentation logic that can be shared between multiple presenters"_. Yes, that's a benefit, but for me the primary aim is encapsulate the behaviour such that we can ask an eligibility: "what was the user response to the QTS question?" and for it to give us an answer _based on the year the claim was made_, without us having to jump through hoops using presenters, and/or helper methods in shared modules. I get that this doesn't follow existing precedence (e.g. we convert boolean values to "yes"/"no" responses in the presenters), but this feels different to those as this encapsulates some complex and important behaviour that brings together: the year the claim was made; the rules for eligibility for the specific policy; the actual stored value representing the users response. My worry is that having this behaviour live in some shared "presentation" code puts it somewhere that isn't immediately obvious, makes it harder to use, and also makes the system more difficult to understand and reason about. 

As a bar minimum the eligibility should be able to report the before/after state for the question, plus the resulting year value that is needed to rehydrate the sentence the user chose (i.e. the combination of `qts_award_year` and `year_for_answer` here in this method). I guess we could have a method that returns those two values to represent the answer given, and then have the actual translation into a human-readable string happen in the presenters. That would make for a bit of a weird method though as it would be returning a pretty abstract data structure that will only make sense and ever be use in a very specific context: turning it into the human-readable string.

Reply:
Yeah, I can't think of a better way of representing the answer - as you say, a single-use data structure, which would probably just have an API of "give me a human-readable answer" wouldn't be any more useful really.

It's interesting that this is the first time that we've had to solve a problem like this - reconstructing an answer. Until now we've been able to store the user's answer directly in the database. But it's the first dynamic question, so makes sense.

One suggestion: update the comment to make it clear that the string returned is not just a string "representing" the answer - which could mean something machine-readable - but is in fact a human-readable description of the answer.

Reply:
👍